### PR TITLE
fix(execution-engine): Fold-over-scalar values' wrong lambda

### PR DESCRIPTION
**Pull request by monoid**
Previously, iterator values in fold over scalar inherited tetraplet from
the scalar.  But for security guarantees, they should have `.[N]` lambda
added, where N is an element index.

### CODE REVIEW

**Review by raftedproc — APPROVED** [review]

File: air/src/execution_step/boxed_value/iterable/lambda_result.rs [hunk 1]
```diff
@@ -57,7 +57,9 @@ impl<'ctx> Iterable<'ctx> for IterableLambdaResult {
         }
 
         let jvalue = &self.jvalues[self.cursor];
-        let result = IterableItem::RefRef((jvalue, &self.tetraplet, 0.into()));
+        let mut tetraplet = (*self.tetraplet).clone();
```
Comment (mikevoronov) [inline air/src/execution_step/boxed_value/iterable/lambda_result.rs:60]: Should we do the same for every member for which `Iterable` is implemented?

File: air/tests/test_module/features/tetraplets/security_tetraplets.rs [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 use air::SecurityTetraplet;
 use air_test_utils::prelude::*;
+use pretty_assertions::assert_eq;
```
Comment (mikevoronov) [inline air/tests/test_module/features/tetraplets/security_tetraplets.rs:19]: let's shuffle it

**Review by mikevoronov — APPROVED** [review]
utACK

File: air/tests/test_module/features/tetraplets/security_tetraplets.rs (hunk 2 above)
Comment (monoid) [reply air/tests/test_module/features/tetraplets/security_tetraplets.rs:19]: How can we shuffle it? Both are `other imports from non-std` (not super/crate).

Comment (mikevoronov) [reply air/tests/test_module/features/tetraplets/security_tetraplets.rs:19]: It's good, I misread it.

**Pull request merged**